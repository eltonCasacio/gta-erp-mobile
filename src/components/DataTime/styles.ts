import styled from 'styled-components/native';
import Theme from '../../styles/Theme';

export const Wrapper = styled.View`
  justify-content: center;
  align-items: center;
`;

export const Text = styled.Text`
  color: ${Theme.Colors.white};
  font-size: 40px;
`;
