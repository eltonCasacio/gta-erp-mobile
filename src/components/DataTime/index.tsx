import React, {useState, useEffect} from 'react';
import {Wrapper, Text} from './styles';

import {Weeks, Months} from '../../utils/weeksAndMonths';

const DataTime = () => {
  const [hour, setHour] = useState('');
  const [date, setDate] = useState('');

  setInterval(() => {
    let getHour = new Date().getHours();
    let getMinute = new Date().getMinutes();

    let hourFormated =
      String(getHour).length < 2 ? '0' + String(getHour) : String(getHour);

    let minutesFormated =
      String(getMinute).length < 2
        ? '0' + String(getMinute)
        : String(getMinute);

    setHour(`${hourFormated}:${minutesFormated}`);
  }, 10000);

  useEffect(() => {
    let day = new Date().getDate();
    let week = Weeks[new Date().getDay()];
    let month = Months[new Date().getMonth()];

    setDate(`${week.slice(0, 3)} ${day}/${month.slice(0, 3)}`);
  }, []);

  return (
    <Wrapper>
      <Text>{hour}</Text>
      <Text>{date}</Text>
    </Wrapper>
  );
};

export default DataTime;
