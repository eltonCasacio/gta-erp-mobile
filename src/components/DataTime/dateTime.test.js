import React from 'react';
import renderer from 'react-test-renderer';

import DataTime from '.';

describe('<DataTime />', () => {
  it('should render default layout', () => {
    const tree = renderer.create(<DataTime />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
