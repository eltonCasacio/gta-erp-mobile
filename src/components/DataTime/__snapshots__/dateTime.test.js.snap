// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataTime /> should render default layout 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fafafa",
          "fontSize": 40,
        },
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#fafafa",
          "fontSize": 40,
        },
      ]
    }
  >
    
  </Text>
</View>
`;
